src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -240,18 +217,20 @@ private static MongoUpdate ConvertModifiedEntryToMongoUpdate(string collectionNa
         IBsonDocumentSerializer entitySerializer,
         IUpdateEntry entry)
     {
-        var idProperty = entry.EntityType.GetIdProperty();
-        var idSerializationInfo = GetPropertySerializationInfo(entitySerializer, idProperty);
-
-        var idValue = entry.GetCurrentValue(idProperty);
-        var serializedIdValue = SerializeValue(idSerializationInfo.Serializer, idValue);
+        var primaryKey = entry.EntityType.FindPrimaryKey();
+        BsonValue idValue = ToBsonDocument(entry, entitySerializer, primaryKey.Properties);
+        if (primaryKey.Properties.Count == 1)
+        {
+            // if PK consist of single field - have to unwrap it
+            idValue = idValue["_id"];
+        }
```

Comment:
I wonder if we should just have a routine here that does this so we can use it in other places.

Reply:
I propose `CreateIdFilter` should be refactored to:

    private static FilterDefinition<BsonDocument> CreateIdFilter(
        IBsonDocumentSerializer entitySerializer,
        IUpdateEntry entry)
    {
        var serializedIdValue = SerializePrimaryKey(entry, entitySerializer);
        return Builders<BsonDocument>.Filter.Eq("_id", serializedIdValue);
    }